starter-code/ShoppingListWithDetailView/app/src/main/java/ly/generalassemb/drewmahrt/shoppinglistdetailview/ShoppingListAdapter.java

```diff
@@ -29,28 +29,36 @@ public ShoppingItemViewHolder onCreateViewHolder(ViewGroup parent, int viewType)
     public void onBindViewHolder(final ShoppingItemViewHolder holder, int position) {
 
         final ShoppingItem currentItem = mShoppingItems.get(position);
-
         holder.mNameTextView.setText(currentItem.getName());
-
-        // Add an OnClickListener that launches DetailActivity and passes it the item's ID
-        holder.mNameTextView.setOnClickListener(new View.OnClickListener() {
+        holder.itemView.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                // Get a reference to the MainActivity as a Context
-                Context mainActivity = holder.mNameTextView.getContext();
-
-                // Create the intent
-                Intent intent = new Intent(mainActivity, DetailActivity.class);
-
-                // Add the ID as an extra
-                intent.putExtra(DetailActivity.ITEM_ID_KEY, currentItem.getId());
-
-                // Start the detail activity
-                mainActivity.startActivity(intent);
+                mListener.onItemSelected(mShoppingItems.get(holder.getAdapterPosition()));
```

Comment:
If you pass just the item ID from MainActivity to the detail fragment, then here it would make sense to pass the item ID, rather than its position in this list, from the adapter back to MainActivity